src/entrypoints/recorder.ts

```diff
@@ -326,34 +342,50 @@ function initXhrObserver(cb: networkCallback, win: IWindow, options: Required<Ne
  *  NB PerformanceNavigationTiming extends PerformanceResourceTiming
  *  Here we don't care which interface it implements as both expose `serverTimings`
  */
-const exposesServerTiming = (event: PerformanceEntry): event is PerformanceResourceTiming =>
-    event.entryType === 'navigation' || event.entryType === 'resource'
-
-function prepareRequest(
-    entry: PerformanceResourceTiming,
-    method: string | undefined,
-    status: number | undefined,
-    networkRequest: Partial<CapturedNetworkRequest>,
+const exposesServerTiming = (event: PerformanceEntry | null): event is PerformanceResourceTiming =>
+    !!event && (event.entryType === 'navigation' || event.entryType === 'resource')
+
+function prepareRequest({
+    entry,
+    method,
+    status,
+    networkRequest,
+    isInitial,
+    start,
+    end,
+}: {
+    entry: PerformanceResourceTiming | null
+    method: string | undefined
+    status: number | undefined
+    networkRequest: Partial<CapturedNetworkRequest>
     isInitial?: boolean
-): CapturedNetworkRequest[] {
+    start?: number
+    end?: number
+}): CapturedNetworkRequest[] {
+    start = entry ? entry.startTime : start
```

Comment:
Might save some complexity on line 376 and 383
```suggestion
    start = entry ? entry.startTime : start ?? 0
```

Reply:
Actually ignore me, looks like we want it to be undefined if it is